### ISSUE TITLE
RFC: should screensaver be re-enabled when inside retroarch gui

### ISSUE THREAD

**Issue by rofl0r** [libretro/RetroArch#16703]
as user of a CRT screen, it seems to make sense if the screensaver is only disabled while emulation is running, but not when inside a retroarch gui screen (say the one that pops up when you press the key combo to access savestates, cheats, and so on). it happens at times that i get a phonecall or any other interruption, so i quickly "exit" the emu going to retroarch menu, and sometimes that menu stays open for a far longer time than anticipated, sometimes hours.

once the user gets back and presses a key on his gamepad (e.g. cursor-up) a signal could be sent to the screensaver to turn it off again - because Xorg doesnt process joystick events when it comes to DPMS. without sending the signal, the user would have to use a mouse or keyboard to get it turned off.

an alternative solution could be that retroarch just sends an "i'm interacting!" ping to the desktop environment whenever joystick input is received. that would make the screensaver even kick in when one pauses a game via in-game facilities. otoh this could interfere when watching a long intro without any interaction.

side-note: currently (as of 1.18.0) when the "disable screensaver" option is enabled, retroarch only sends a command to disable the screensaver, but not to turn it back on when exiting.

anyway, please share your thoughts, so once we can find an agreement i can start to look into the implementation.

### Environment information

- OS: linux/X11

**Comment by hizzlekizzle** [comment]
Well, RGUI--which is the CRT-focused menu--has the option to fade out the static elements and leave only the moving backgrounds after a few minutes.

**Comment by rofl0r** [comment]
@hizzlekizzle 
while not optimal, that's still a pretty good solution. i found the rather well hidden option in rgui and it works good enough to prevent permanent screen damage. since there's no other input i guess we can close this and accept this as a workaround.

**Comment by Apteryks** [comment]
Does RetroArch still not re-enable screensaver when exiting the whole RetroArch application? It seems to happen this way, using GNOME; my screensaver appears to remains disabled (that is, it doesn't kick in after the configured 5 minutes of inactivity) after using RetroArch with the 'disable screensaver' option on.

**Comment by rofl0r** [comment]
> Does RetroArch still not re-enable screensaver when exiting the whole RetroArch application?

unless it was added after 1.18.0, then no (i didn't). this should probably be a separate issue report though.

**Comment by Apteryks** [comment]
@rofl0r I've investigated a bit and according to `gfx/common/dbus_common.c`, `dbus_screensaver_uninhibit` should take care of that. With the following log added:
```diff
modified   gfx/common/dbus_common.c
@@ -144,8 +144,10 @@ void dbus_screensaver_uninhibit(void)
          DBUS_TYPE_UINT32, &dbus_screensaver_cookie,
          DBUS_TYPE_INVALID);
 
-   if (dbus_connection_send(dbus_connection, msg, NULL))
+   if (dbus_connection_send(dbus_connection, msg, NULL)) {
       dbus_connection_flush(dbus_connection);
+      RARCH_LOG("[DBus]: Uninhibited screensaver via DBus.\n");
+   }
    dbus_message_unref(msg);
 
    dbus_screensaver_cookie = 0;
```
I could see this upon exiting RetroArch:
```
[INFO] [PulseAudio]: Pausing.
[INFO] [Core]: Unloading core..
[INFO] [Core]: Unloading core symbols..
[INFO] [DBus]: Uninhibited screensaver via DBus.
```

**Comment by rofl0r** [comment]
then the question remains why your screensaver stays disabled. note that on my system there is no dbus and i used X11-related screensaver inhibiting code, and at the time i wrote #16702 it stayed inhibited after terminating too. either the uninhibit code isn't called at exit or the inhibit code is called multiple times during program execution and the uninhibiter only once or a similar issue. i had in mind to investigate that at the time but didn't get around to do it.

**Comment by Apteryks** [comment]
`gnome-session-inhibit --list` suggests there is no problem with RetroArch, if I can trust that (when RetroArch exits, it vanishes from the list).
